controllers/welcome.rb

```diff
@@ -1,3 +1,89 @@
+# ===== welcome
 get '/' do
+  @apartments = Apartment.all
   erb :welcome
 end
+
+# ===== ===== ===== APARTMENTS ===== ===== =====
+# ===== show apartment
+get '/apartments/:id' do
+  @apartment = Apartment.find(@params[:id])
+  erb :apartment/show
+end
+
+# ===== edit apartment
+get '/apartments/:id/edit' do
+  @apartment = Apartment.find(@params[:id])
+  erb :apartment/edit
+end
+
+# ===== update apartment
+put '/apartments/:id/update' do
+  @apartment = Apartment.find(@params[:id])
+  @apartment.update(address: params[:address], monthly_rent: params[:monthly_rent], sqft: params[:sqft], num_beds: params[:num_beds], num_baths: params[:num_baths])
+  redirect "/apartments"
+end
+
+# ===== new apartment
+get '/apartments/new' do
+  erb :apartment/new
+end
+
+# ===== create apartment
+post '/apartments/create' do
+  Apartment.create(address: params[:address], monthly_rent: params[:monthly_rent], sqft: params[:sqft], num_beds: params[:num_beds], num_baths: params[:num_baths])
+  redirect "/apartments"
+end
+
+# ===== delete apartment
+delete "apartments/:id" do
+  @apartment = Apartment.find(@params[:id])
+  @apartment.destroy
+  redirect "/apartments"
+end
+
+# ===== ===== ===== TENENTS ===== ===== =====
+# ===== see tenents for selected apartment
+get '/apartment/:id/tenents' do
+  @apartment = Apartment.find(@params[:id])
+  @tenents = Tenent.where(apartment_id: @params[:id])
```

Comment:
Technically, you could shorten this using the helper methods given to you by active record:
This line is equivalent to:
`@tenents = @apartment.tenants`